File: basemodule/src/main/java/com/android/base/utils/ExitAppUtils.java
package com.android.base.utils;

import android.app.Activity;

import java.util.LinkedList;
import java.util.List;

public class ExitAppUtils {

	private List<Activity> mActivityList = new LinkedList<Activity>();
	private static ExitAppUtils instance = new ExitAppUtils();

	private ExitAppUtils(){};

	public static ExitAppUtils getInstance(){
		return instance;
	}

	public void addActivity(Activity activity){
		mActivityList.add(activity);
	}

	public void delActivity(Activity activity){
		mActivityList.remove(activity);
	}

	public void exit(){
		for(Activity activity : mActivityList){
			activity.finish();
		}
		
		System.exit(0);
	}
	

}

File: AndroidLearnProject/basemodule/src/main/java/com/android/learn/base/utils/ExitAppUtils.kt
package com.android.learn.base.utils

import android.app.Activity

import java.util.LinkedList

class ExitAppUtils private constructor() {

    private val mActivityList = LinkedList<Activity>()

    fun addActivity(activity: Activity) {
        mActivityList.add(activity)
    }

    fun delActivity(activity: Activity) {
        mActivityList.remove(activity)
    }

    fun exit() {
        for (activity in mActivityList) {
            activity.finish()
        }

        System.exit(0)
    }

    companion object {
        val instance = ExitAppUtils()
    }


}